data-mate/ utils string functions

ciorg opened the pull request:
* isPort
* isMimeType
* isISDN
* isPhoneNumberLIke
* isPostolCode
* isAlphaNumeric

peterdemartini commented on packages/data-mate/src/function-configs/string/isMimeType.ts:
```diff
@@ -0,0 +1,11 @@
+import { FieldValidateConfig, ProcessMode, FunctionDefinitionType } from '../interfaces';
+
+export const isMimeTypeConfig: FieldValidateConfig = {
+    name: 'isMimeType',
```
Shouldn't this be `isMIMEType`?

ciorg replied:
fixed this with latest commit

peterdemartini commented on packages/data-mate/src/function-configs/string/isMimeType.ts:
```diff
@@ -0,0 +1,11 @@
+    name: 'isMimeType',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    description: 'Checks to see if input is a MIME Type',
```
This description doesn't give you any more value than the function itself. Maybe describe what MIME is?

ciorg replied:
I added more details to the description

peterdemartini commented on packages/data-mate/src/function-configs/string/isMimeType.ts:
```diff
@@ -0,0 +1,11 @@
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    description: 'Checks to see if input is a MIME Type',
+    create() { return isMimeType; },
+    accepts: [],
```
This should probably only accept strings since this could be optimized out if given anything else
```suggestion
    accepts: [FieldType.String],
```

ciorg replied:
fixed this with latest commit

peterdemartini commented on packages/utils/src/phone-number.ts:
```diff
@@ -13,3 +15,21 @@ export function parsePhoneNumber(input: string|number): string {
+    return false;
+}
+
+export function isPhoneNumberLike(input: unknown): boolean {
```
What is the benefit of this function? It seems extremely loose?

ciorg replied:
It is extremely loose, the problem is the isISDN assumes everything is in an international phone number format and throws out a lot of possibly good phone numbers. 

If a country is provided along with a number to the ISDN it becomes much more reliable, and thinking about this the country needs to be an optional arg for the isISDN function.  

But in enough cases the data didn't have accurate country info and isISDN was rejecting most of the phone number values, so the isPhoneNumber like function was created.  A basic check based on number of  digits, min and max for a valid phone number.

peterdemartini replied:
We should probably have that described in jsdoc

ciorg commented:
Had a strange issue where I changed the file isMimeType -> isMIMEType and was causing an error on the build saying it couldn't find the isMIMEType file for the CI tests.  It built fine locally, but I changed the file name back to isMimeType and it passed.  Not sure  whats going on there.

peterdemartini commented:
That's because macos file system is case insensitive by default. You need to use `git mv` to tell git the file was moved. You might need to rename it back and then use the `git mv` command.

peterdemartini commented on packages/data-mate/src/function-configs/string/isAlphaNumeric.ts:
```diff
@@ -0,0 +1,33 @@
+    validate_arguments({ locale }) {
+        if (locale == null) return;
+
+        if (!isString(locale) || !('this' in validator.isAlphanumericLocales)) {
```
```suggestion
        if (!isString(locale) || !(locale in validator.isAlphanumericLocales)) {
```

peterdemartini commented:
This needs to be updated from master, also we should be writing examples (a lot of the spec files you created can be deleted)

jsnoble commented on packages/data-mate/src/function-configs/string/isBase64.ts:
```diff
@@ -58,5 +58,7 @@ export const isBase64Config: FieldValidateConfig = {
     create() { return isBase64; },
-    accepts: [FieldType.String],
+    accepts: [
+        FieldType.String
```
does this need to be on a new line?

jsnoble commented on packages/data-mate/src/function-configs/string/isCountryCode.ts:
```diff
@@ -86,5 +86,7 @@ export const isCountryCodeConfig: FieldValidateConfig = {
     create() { return isCountryCode; },
-    accepts: [FieldType.String],
+    accepts: [
+        FieldType.String
```
does this need to be on a new line?

jsnoble commented on packages/data-mate/src/function-configs/string/isEmail.ts:
```diff
@@ -144,5 +144,7 @@ export const isEmailConfig: FieldValidateConfig = {
     },
-    accepts: [FieldType.String],
+    accepts: [
+        FieldType.String
```
does this need to be on a new line?

jsnoble commented on packages/data-mate/src/function-configs/string/isFQDN.ts:
```diff
@@ -114,5 +114,7 @@ export const isFQDNConfig: FieldValidateConfig = {
     create() { return isFQDN; },
-    accepts: [FieldType.String],
+    accepts: [
+        FieldType.String
```
does this need to be on a new line?

jsnoble reviewed (changes requested):
Is there a new reason for the styling changes

jsnoble reviewed (approved):
LTGM
